core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeAstBuilder.scala

```diff
@@ -56,31 +55,63 @@ class SnowflakeAstBuilder extends SnowflakeParserBaseVisitor[TreeNode] {
   }
 }
 
-class SnowflakeRelationBuilder extends SnowflakeParserBaseVisitor[Relation] {
-  override def visitObject_ref(ctx: SnowflakeParser.Object_refContext): Relation = {
+class SnowflakeRelationBuilder extends SnowflakeParserBaseVisitor[ir.Relation] {
+  override def visitObject_ref(ctx: SnowflakeParser.Object_refContext): ir.Relation = {
     val tableName = ctx.object_name().id_(0).getText
-    NamedTable(tableName, Map.empty, is_streaming = false)
+    ir.NamedTable(tableName, Map.empty, is_streaming = false)
   }
+
+  override def visitTable_source_item_joined(ctx: Table_source_item_joinedContext): ir.Relation = {
+
+    def translateJoinType(joinType: SnowflakeParser.Join_typeContext): ir.JoinType = {
+      if (joinType == null || joinType.outer_join() == null) {
+        ir.InnerJoin
+      } else if (joinType.outer_join().LEFT() != null) {
+        ir.LeftOuterJoin
+      } else if (joinType.outer_join().RIGHT() != null) {
+        ir.RightOuterJoin
+      } else if (joinType.outer_join().FULL() != null) {
+        ir.FullOuterJoin
+      } else {
+        ir.UnspecifiedJoin
+      }
+    }
+
+    def buildJoin(left: ir.Relation, right: SnowflakeParser.Join_clauseContext): ir.Join = {
+
+      ir.Join(
+        left,
+        right.object_ref().accept(this),
+        None,
+        translateJoinType(right.join_type()),
+        Seq(),
+        ir.JoinDataType(is_left_struct = false, is_right_struct = false))
+    }
+    val left = ctx.object_ref().accept(this)
+    ctx.join_clause().asScala.foldLeft(left)(buildJoin)
+  }
+
 }
-class SnowflakeExpressionBuilder extends SnowflakeParserBaseVisitor[Expression] {
 
-  override def visitSelect_list_elem(ctx: SnowflakeParser.Select_list_elemContext): Expression = {
+class SnowflakeExpressionBuilder extends SnowflakeParserBaseVisitor[ir.Expression] {
```

Comment:
@vil1 for the next PRs - split the `SnowflakeExpressionBuilder` into a separate file